src/Npgsql/NpgsqlConnector.cs

```diff
@@ -599,7 +599,12 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
         void Connect(NpgsqlTimeout timeout)
         {
             EndPoint[] endpoints;
-            if (!string.IsNullOrEmpty(Host) && Host[0] == '/')
+
+            // Unix-domain sockets are assumed when the host name:
+            // - starts with a forward slash (rooted paths on Unix, e. g. "/tmp")
+            // - has a colon as second character (rooted classical paths on Windows, e. g. "C:\my\socket\dir")
+            // - starts with a backslash (long paths or UNC paths on Windows, e. g. "\\?\D:\very long path" or "\\server\share")
+            if (!string.IsNullOrEmpty(Host) && (Host[0] == '/' || Host[1] == ':' || Host[0] == '\\'))
```

Comment:
```suggestion
            if (Path.IsPathRooted(Host))
```

Reply:
Oh nice, didn't know about this! Worth just looking inside to make sure this won't accidentally consider any valid hostname/IP as rooted.

Reply:
It does the same thing which Brar wrote, but without a bug. The current code doesn't take into account that the host name could be one character length and attempt to access a value at index 1 which will cause `ArgumentOutOfRangeException`.

Reply:
> It does the same thing which Brar wrote, but without a bug.

It does indeed:

- https://github.com/dotnet/runtime/blob/386ee4658291003d1d4fe8e28612d9c7927f5d40/src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs#L190
- https://github.com/dotnet/runtime/blob/386ee4658291003d1d4fe8e28612d9c7927f5d40/src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs#L106